src/basic/strv.h

```diff
@@ -35,18 +35,32 @@ size_t strv_length(char * const *l) _pure_;
 int strv_extend_strv(char ***a, char * const *b, bool filter_duplicates);
 int strv_extend_strv_concat(char ***a, char * const *b, const char *suffix);
 int strv_prepend(char ***l, const char *value);
-int strv_extend(char ***l, const char *value);
+
+int strv_extend_with_size(char ***l, size_t *n, const char *value);
```

Comment:
Can you add a brief comment here, replicating the line about speed? Otherwise it won't be super obvious to someone else some time in the future